storage/backend/gcs/gcs.go

```diff
@@ -140,6 +140,41 @@ func (b *Backend) Put(ctx context.Context, p string, r io.Reader) error {
 	}
 }
 
+// Exists checks if object already exists.
+func (b *Backend) Exists(ctx context.Context, p string) (bool, error) {
+	errCh := make(chan error)
```

Comment:
Although I'm not completely sure if it'd improve anything, we can introduce an inline result struct `result{bool, err}` and use only a single channel to collect results. What do you think?